R/derive_param_confirmed_bor.R

```diff
@@ -0,0 +1,528 @@
+#'
+#'   1. For each subject the best response as derived in the previous step is
+#'   selected, where `"CR"` is best and `"MISSING"` is worst in the order above.
+#'   If the best response is not unique, the first one (with respect to `ADT`)
+#'   is selected. Only the variables `AVALC`, `ADT`, and the variables specified
+#'   for `subject_keys` are kept from the selected observations.
+#'
+#'   1. The `AVAL` variable is added and set to `aval_fun(AVALC)`.
+#'
+#'   1. The variables specified by the `set_values_to` parameter are added to
+#'   the new observations.
+#'
+#'   1. The new observations are added to input dataset.
+#'
+#' @return The input dataset with a new parameter for confirmed best overall
+#'   response
+#'
+#' @keywords derivation adrs
+#'
+#' @author Stefan Bundfuss
+#'
+#' @export
+#'
+#' @examples
+#'
+#' library(dplyr)
+#'
+#' # Create ADSL dataset
+#' adsl <- tibble::tribble(
+#'   ~USUBJID, ~TRTSDTC,
+#'   "1",      "2020-01-01",
+#'   "2",      "2019-12-12",
+#'   "3",      "2019-11-11",
+#'   "4",      "2019-12-30",
+#'   "5",      "2020-01-01",
+#'   "6",      "2020-02-02",
+#'   "7",      "2020-02-02",
+#'   "8",      "2020-04-01"
+#' ) %>%
+#'   mutate(
+#'     TRTSDT = lubridate::ymd(TRTSDTC),
+#'     STUDYID = "XX1234"
+#'   )
+#'
+#' # Create ADRS dataset
+#' ovr_obs <- tibble::tribble(
+#'   ~USUBJID, ~ADTC,        ~AVALC,
+#'   "1",      "2020-01-01", "PR",
+#'   "1",      "2020-02-01", "CR",
+#'   "1",      "2020-02-16", "NE",
+#'   "1",      "2020-03-01", "CR",
+#'   "1",      "2020-04-01", "SD",
+#'   "2",      "2020-01-01", "SD",
+#'   "2",      "2020-02-01", "PR",
+#'   "2",      "2020-03-01", "SD",
+#'   "2",      "2020-03-13", "CR",
+#'   "3",      "2019-11-12", "CR",
+#'   "3",      "2019-12-02", "CR",
+#'   "3",      "2020-01-01", "SD",
+#'   "4",      "2020-01-01", "PR",
+#'   "4",      "2020-03-01", "SD",
+#'   "4",      "2020-04-01", "SD",
+#'   "4",      "2020-05-01", "PR",
+#'   "4",      "2020-05-15", "NON-CR/NON-PD",
+#'   "5",      "2020-01-01", "PR",
+#'   "5",      "2020-01-10", "SD",
+#'   "5",      "2020-01-20", "PR",
+#'   "5",      "2020-05-15", "NON-CR/NON-PD",
+#'   "6",      "2020-02-06", "PR",
+#'   "6",      "2020-02-16", "CR",
+#'   "6",      "2020-03-30", "PR",
+#'   "6",      "2020-04-12", "PD",
+#'   "6",      "2020-05-01", "CR",
+#'   "6",      "2020-06-01", "CR",
+#'   "7",      "2020-02-06", "PR",
+#'   "7",      "2020-02-16", "CR",
+#'   "7",      "2020-04-01", "NE"
+#' ) %>%
+#'   mutate(PARAMCD = "OVR")
+#'
+#' pd_obs <-
+#'   bind_rows(tibble::tribble(
+#'     ~USUBJID, ~ADTC,        ~AVALC,
+#'     "6",      "2020-04-12", "Y"
+#'   ) %>%
+#'     mutate(PARAMCD = "PD"))
+#' adrs <- bind_rows(ovr_obs, pd_obs) %>%
+#'   mutate(
+#'     ADT = lubridate::ymd(ADTC),
+#'     STUDYID = "XX1234"
+#'   ) %>%
+#'   select(-ADTC) %>%
+#'   admiral::derive_vars_merged(
+#'     dataset_add = adsl,
+#'     by_vars = dplyr::vars(STUDYID, USUBJID),
```

Comment:
why do we include dplyr:: or lubridate:: here. this is what Stephen was requesting but i thought it wasn't our convention. is it because we have our own admiral versions of the same function name? if so anyway to avoid repeating tidyverse function names in admiral?

Reply:
These are just left overs from the testing phase. I have removed them.